test/sagemaker_tests/huggingface_tensorflow/training/integration/conftest.py

```diff
@@ -133,17 +133,13 @@ def instance_type(request, processor):
 
 
 @pytest.fixture()
-def py_version():
-    if "TEST_PY_VERSIONS" in os.environ:
-        return os.environ["TEST_PY_VERSIONS"].split(",")
-    return None
+def py_version(request):
+    return request.config.getoption("--py-version")
 
 
 @pytest.fixture()
-def processor():
-    if "TEST_PROCESSORS" in os.environ:
-        return os.environ["TEST_PROCESSORS"].split(",")
-    return None
+def processor(request):
+    return request.config.getoption("--processor")
```

Comment:
The previous return value of `processor` fixture was `["gpu"]` when `--processor gpu` was used as an input arg, because the return value of `"gpu".split(",")` is a list.

Therefore, `is_gpu = processor == "gpu"` as used in the `skip_by_device_type` fixture would be `False`, leading to tests being skipped on GPU DLCs in GPU instances.